src/js/contentUtils.ts

```diff
@@ -132,10 +132,11 @@ export function storeFoundJS(scriptNodeMaybe: HTMLScriptElement): void {
 
     let rawManifest: RawManifest | null = null;
     try {
-      rawManifest = JSON.parse(scriptNodeMaybe.textContent!);
-      if (!rawManifest) {
-        throw new Error('rawManifest is null or empty');
+      if (scriptNodeMaybe.textContent == null) {
```

Comment:
I waffled a bit on this block and ended up leaving it as is because I was unsure if it's possible for a node's content to go from null to non-null upon retrying.

If it is not (which is what you are implying here?) I would lean toward moving this check before `try` and using the `invalidateAndThrow` method. In general I think we should follow this pattern of throwing whenever we invalidate to avoid weird things happening after the invalidation happens.

Reply:
Addressed